tests/integration/repo_webhook_test.go

```diff
@@ -1102,45 +1105,465 @@ type workflowRunWebhook struct {
+
+	repoName := "test-workflow-run-cancelling-abandoned-run"
+	testRepo := unittest.AssertExistsAndLoadBean(t, &repo.Repository{ID: createActionsTestRepo(t, token, repoName, false).ID})
+
+	runners := make([]*mockRunner, 2)
+	for i := range runners {
+		runners[i] = newMockRunner()
+		runners[i].registerAsRepoRunner(t, "user2", repoName,
+			fmt.Sprintf("mock-runner-%d", i), []string{"ubuntu-latest"}, false)
+	}
+
+	testAPICreateWebhookForRepo(t, session, "user2", repoName, webhookData.URL, "workflow_run")
+
+	ctx := t.Context()
+	gitRepo, err := gitrepo.OpenRepository(ctx, testRepo)
+	assert.NoError(t, err)
+
+	// 2.2 trigger the webhooks
+
+	// add workflow file to the repo
+	// init the workflow
+	wfilename := "push.yml"
+	wfTreePath := ".gitea/workflows/" + wfilename
+	wfFileContent := `on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test2:
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test3:
+    needs: [test, test2]
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test4:
+    needs: [test, test2, test3]
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test5:
+    needs: [test, test2, test4]
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test6:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04, ubuntu-24.04]
+    needs: [test, test2, test3]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - run: exit 0
+
+  test7:
+    needs: test6
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test8:
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+
+  test9:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04, ubuntu-24.04, ubuntu-25.04, windows-2022, windows-2025, macos-13, macos-14, macos-15]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - run: exit 0
+
+  test10:
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0`
+
+	opts := getWorkflowCreateFileOptions(user2, testRepo.DefaultBranch, "create "+wfTreePath, wfFileContent)
+	createWorkflowFile(t, token, "user2", repoName, wfTreePath, opts)
+
+	commitID, err := gitRepo.GetBranchCommitID(testRepo.DefaultBranch)
+	assert.NoError(t, err)
+
+	// 3. validate the webhook is triggered
+	assert.Equal(t, "workflow_run", webhookData.triggeredEvent)
+	assert.Len(t, webhookData.payloads, 1)
+	assert.Equal(t, "requested", webhookData.payloads[0].Action)
+	assert.Equal(t, "queued", webhookData.payloads[0].WorkflowRun.Status)
+	assert.Equal(t, testRepo.DefaultBranch, webhookData.payloads[0].WorkflowRun.HeadBranch)
+	assert.Equal(t, commitID, webhookData.payloads[0].WorkflowRun.HeadSha)
+	assert.Equal(t, repoName, webhookData.payloads[0].Repo.Name)
+	assert.Equal(t, "user2/"+repoName, webhookData.payloads[0].Repo.FullName)
+
+	for _, runner := range runners {
+		task := runner.fetchTask(t)
+		if !allJobsAbandoned {
```

Comment:
I prefer to change this to
```suggestion
	if !allJobsAbandoned {
		for _, runner := range runners {
			task := runner.fetchTask(t)
```
Not fetching any job via fetchTask means, abandon them via the setting I decremented to 0 in this test.

Otherwise you would have to set endless task timeout to zero as well so another cleanup function cancels.

since https://github.com/go-gitea/gitea/pull/35215#discussion_r2293193505 reverted [my change](https://github.com/go-gitea/gitea/pull/35215/commits/d00d5cfe21c02d28bef0cead3f19db038dfa601f)